Sources/Yams/Constructor.swift

```diff
@@ -0,0 +1,360 @@
+        guard let scalar = node.scalar else { fatalError("Never happen this") }
+        return scalar
+    }
+}
+
+// MARK: - Types that can't conform to ScalarConstructible
+extension NSNull/*: ScalarConstructible*/ {
+    public static func construct(from node: Node) -> NSNull? {
+        guard let scalar = node.scalar else { fatalError("Never happen this") }
+        switch scalar {
+        case "", "~", "null", "Null", "NULL":
+            return NSNull()
+        default:
+            return nil
+        }
+    }
+}
+
+// MARK: mapping
+extension Dictionary {
+    public static func construct_mapping(from node: Node) -> [AnyHashable: Any]? {
+        return _construct_mapping(from: node)
+    }
+
+    fileprivate static func _construct_mapping(from node: Node) -> [AnyHashable: Any] {
+        guard let pairs = flatten_mapping(node).pairs else { fatalError("Never happen this") }
+        var dictionary = [AnyHashable: Any](minimumCapacity: pairs.count)
+        pairs.forEach {
+            // TODO: YAML supports keys other than str.
+            dictionary[String._construct(from: $0.key)] = node.tag.constructor.any(from: $0.value)
+        }
+        return dictionary
+    }
+
+    private static func flatten_mapping(_ node: Node) -> Node {
+        guard var pairs = node.pairs else { fatalError("Never happen this") }
+        var merge = [Pair<Node>]()
+        var index = pairs.startIndex
+        while index < pairs.count {
+            let pair = pairs[index]
+            if pair.key.tag.name == .merge {
+                pairs.remove(at: index)
+                switch pair.value {
+                case .mapping:
+                    let flattened_node = flatten_mapping(pair.value)
+                    if let pairs = flattened_node.pairs {
+                        merge.append(contentsOf: pairs)
+                    }
+                case let .sequence(array, _):
+                    let submerge = array
+                        .filter { $0.isMapping } // TODO: Should raise error on other than mapping
+                        .flatMap { flatten_mapping($0).pairs }
+                        .reversed()
+                    submerge.forEach {
+                        merge.append(contentsOf: $0)
+                    }
+                default:
+                    break // TODO: Should raise error on other than mapping or sequence
+                }
+            } else if pair.key.tag.name == .value {
+                pair.key.tag.name = .str
+                index += 1
+            } else {
+                index += 1
+            }
+        }
+        return .mapping(merge + pairs, node.tag)
+    }
+}
+
+extension Set {
+    public static func construct_set(from node: Node) -> Set<AnyHashable>? {
+        guard let pairs = node.pairs else { fatalError("Never happen this") }
+        // TODO: YAML supports Hashable elements other than str.
+        return Set<AnyHashable>(pairs.map({ String._construct(from: $0.key) as AnyHashable }))
```

Comment:
This is required.
This static function is defined in `extension Set {}`.
Swift compiler can't infer the type of generic parameter at here.
```swift
extension Set {
    public static func takes(array: [AnyHashable]) -> Set<AnyHashable> {
        return Set(array)
//    Playground execution failed: error: 104351.xcplaygroundpage:3:16:
//    error: expression type 'Set<Element>' is ambiguous without more context
//    return Set(array)
    }
}
```